graylog2-web-interface/src/logic/telemetry/TelemetryProvider.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React, { useMemo, useEffect, useState } from 'react';
+import { usePostHog } from 'posthog-js/react';
+
+import type { TelemetryEventType, TelemetryEvent } from 'logic/telemetry/TelemetryContext';
+import TelemetryContext from 'logic/telemetry/TelemetryContext';
+import { TelemetrySettingsActions } from 'stores/telemetry/TelemetrySettingsStore';
+import TelemetryInfoModal from 'logic/telemetry/TelemetryInfoModal';
+import type { TelemetryDataType } from 'logic/telemetry/useTelemetryData';
+import useTelemetryData from 'logic/telemetry/useTelemetryData';
+
+const TelemetryProvider = ({ children }: { children: React.ReactElement }) => {
+  const posthog = usePostHog();
+  const { data: telemetryData, isSuccess: isTelemetryDataLoaded, refetch: refetchTelemetryData } = useTelemetryData();
+  const [showTelemetryInfo, setShowTelemetryInfo] = useState<boolean>(false);
+
+  useEffect(() => {
+    const setGroup = () => {
+      if (isTelemetryDataLoaded
+        && telemetryData
+        && telemetryData.user_telemetry_settings?.telemetry_enabled) {
+        const {
+          cluster: { cluster_id: clusterId, ...clusterDetails },
+          current_user: { user, ...userDetails },
+          license,
+          plugin,
+          search_cluster: searchCluster,
+          user_telemetry_settings: { telemetry_permission_asked: isPermissionAsked },
+        } = telemetryData as TelemetryDataType;
+
+        posthog.group('cluster', clusterId, {
+          ...clusterDetails,
+          ...license,
+          ...plugin,
+          ...searchCluster,
+        });
+
+        posthog.identify(user, { ...userDetails });
+        setShowTelemetryInfo(!isPermissionAsked);
+      }
+    };
+
+    if (posthog) {
+      setGroup();
+    }
+  }, [posthog, isTelemetryDataLoaded, telemetryData]);
+
+  const TelemetryContextValue = useMemo(() => {
+    const sendTelemetry = (eventType: TelemetryEventType, event: TelemetryEvent) => {
+      if (posthog) {
+        posthog.capture(eventType, event);
```

Comment:
What happens when the user has telemetry disabled? We are skipping the group configuration in the `useEffect` block in that case, but should we also just return a no-op function here instead of calling `posthog.capture()`?

Reply:
Thank you, @dennisoelkers, for having a look and for the suggestions.

> What happens when the user has telemetry disabled? We are skipping the group configuration in the useEffect block in that case, but should we also just return a no-op function here instead of calling posthog.capture()?


When telemetry is disabled, the init is not called, and the PostHog context should be empty, resulting in `usePostog` returning `posthog` with the value of `undefined`, therefor calling `posthog.capture` is not called.